linux_os/guide/services/ntp/ntpd_run_as_ntp_user/oval/shared.xml

```diff
@@ -0,0 +1,36 @@
+<def-group>
+  <definition class="compliance" id="ntpd_run_as_ntp_user" version="1">
+    <metadata>
+      <title>Check if ntpd is configured to run under ntp user account</title>
+      {{{- oval_affected(products) }}}
+      <description>check two files which can set ntpd command line options for correct values</description>
+    </metadata>
+    <criteria operator="OR">
+      <criterion comment="check /etc/sysconfig/ntpd for correct value" test_ref="test_ntpd_run_as_ntp_user_etc_sysconfig_ntpd" />
+      <criterion comment="check /usr/lib/systemd/system/ntpd.service for correct value" test_ref="test_ntpd_run_as_ntp_user_systemd" />
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test id="test_ntpd_run_as_ntp_user_etc_sysconfig_ntpd" version="1" check="all" comment="check /etc/sysconfig/ntpd">
+    <ind:object object_ref="obj_ntpd_run_as_ntp_user_etc_sysconfig_ntpd" />
+  </ind:textfilecontent54_test>
+
+    <ind:textfilecontent54_object id="obj_ntpd_run_as_ntp_user_etc_sysconfig_ntpd"
+  version="1" comment="check /etc/sysconfig/ntpd">
+    <ind:filepath>/etc/sysconfig/ntpd</ind:filepath>
+    <ind:pattern operation="pattern match">^[\s]*OPTIONS=.*-u ntp:ntp.*$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+
+  <ind:textfilecontent54_test id="test_ntpd_run_as_ntp_user_systemd" version="1" check="all" comment="check /usr/lib/systemd/system/ntpd.service">
```

Comment:
Why not use the systemd probes?

Reply:
This is a valid question, it did not occur to me that I could use systemd probes. However, I did some investigation and found out that rewriting this single test to use systemd_unit_property is really not a trivial task, especially when talking about ExecStart property. Therefore I suggest that now we leave as it is, because technically it is doing the right thing - checking the unit file. I will open an issue that  similar tests should be rewritten from filecontent tests to systemd_unit_property tests in future as. Is this acceptable for you?

Reply:
@vojtapolasek Can you explain why do you think it's difficult? I think you can get the ExecStart property and then compare it against a regex in an OVAL state. What is the catch?

Reply:
Well the catch is that if you do for example:
$ systemctl show cups.service
You will get array of various values, not just the ExecStart line.
According to OVAL documentation the output of this probe is the same as running
$ systemctl show <unit>
As I said, this should be changed in future for all checks checking for systemd unit files, but it will not be a trivial work.

Reply:
It should be something actually quite trivial. For example:
````
  <linux:systemdunitproperty_test check="all" check_existence="all_exist" id="test_execstart" comment="test" version="1">
    <linux:object object_ref="obj_execstart" />
    <linux:state state_ref="state_execstart" />
  </linux:systemdunitproperty_test>
  <linux:systemdunitproperty_object id="obj_execstart" version="1">
    <linux:unit>ntpd.service</linux:unit>
    <linux:property>ExecStart</linux:property>
  </linux:systemdunitproperty_object>
  <linux:systemdunitproperty_state id="state_execstart" version="1">
    <linux:unit>ntpd.service</linux:unit>
    <linux:property>ExecStart</linux:property>
    <linux:value>some_value</linux:value>
  </linux:systemdunitproperty_state>
```
However, `ExecStart` is not being gotten by the probe like it should. @jan-cerny is this expected?

Reply:
@vojtapolasek You should be able to get only the ExecStart by specifying `ExecStart` in `property` element as @redhatrises has shown. The problem is that this doesn't work in OpenSCAP. I have tried it and the probe simply doesn't return this property. I think it's a bug. I will check it and report this problem as a BZ. Until it's fixed we have to use the textfilecontent probe to read the unit file.

Reply:
The problem has been reported as a bug in https://bugzilla.redhat.com/show_bug.cgi?id=1819773.

I think that now we can stay with the check proposed in this PR.